File: easybluetoothble/src/main/java/com/srise/easybluetoothble/impl/SendMessageBusiness.java
package com.srise.easybluetoothble.impl;

import android.util.Log;

import com.srise.easybluetoothble.Interface.ISendDataEngine;
import com.srise.easybluetoothble.Interface.ISendMessage;
import com.srise.easybluetoothble.bean.BaseMessage;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;


public class SendMessageBusiness implements ISendMessage {
    private final static String TAG = "SendMessageBusiness";

    private ISendDataEngine mSendEngine;
    private ExecutorService mExecutorService;

    public SendMessageBusiness() {
        mSendEngine = new BleSendDataEngine();
        mExecutorService = Executors.newSingleThreadExecutor();
    }

    @Override
    public void sendMessage(BaseMessage message) {
        Log.d(TAG , " sendMessage");
        mExecutorService.submit(new SendTask(message));
    }

    public void destroy() {
        Log.d(TAG , " destroy");
        mExecutorService.shutdownNow();
    }

    private class SendTask implements Runnable {
        private BaseMessage mMessage;

        public SendTask(BaseMessage message) {
            mMessage = message;
        }

        @Override
        public void run() {
            Log.d(TAG , " SendTask run");

            mSendEngine.sendData(mMessage);
        }
    }
}

File: ktlionble/src/main/java/c/example/dell/ktlionble/impl/SendMessageBusiness.kt
package c.example.dell.ktlionble.impl

import android.util.Log
import c.example.dell.ktlionble.bean.BaseMessage
import c.example.dell.ktlionble.inter.ISendDataEngine
import c.example.dell.ktlionble.inter.ISendMessage
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors;

class SendMessageBusiness : ISendMessage {
    companion object {
        const val TAG = "SendMessageBusiness"
    }

    private val mSendEngine: ISendDataEngine? = BleSendDataEngine()
    private val mExecutorService: ExecutorService? = Executors.newSingleThreadExecutor()

    override fun sendMessage(message: BaseMessage) {
        Log.d(TAG, " sendMessage")
        mExecutorService!!.submit(SendTask(message))
    }

    override fun destroy() {
        Log.d(TAG, " destroy");
        mExecutorService!!.shutdownNow();
    }

    inner class SendTask() : Runnable { //TODO
        private var mMessage: BaseMessage? = null

        constructor(message: BaseMessage) : this() {
            mMessage = message
        }

        override fun run() {
            Log.d(TAG, " SendTask run")
            mSendEngine!!.sendData(mMessage!!)
        }
    }
}